spec/core_spec.rb

```diff
@@ -1018,4 +1018,89 @@ class User3Serializer < BaseTestSerializer
       include_examples 'lazy loader when include_data option is set'
     end
   end
+
+  describe "using serialization scope" do
+    class BlogPostsLoader < AmsLazyRelationships::Loaders::Base
+      def load_data(records, loader, scope)
+        data = []
+
+        records.each do |r|
+          d = r.blog_posts
+          d = d.where(title: scope[:title]) if scope
+
+          loader.call(r, d)
+          data << d
+        end
+
+        data.flatten.compact.uniq
+      end
+
+      def batch_key(_)
+        "Key"
+      end
+    end
+
+    let(:level0_serializer_class) do
+      class Level1Serializer11 < BaseTestSerializer
```

Comment:
That should be `Level1Serializer16` I believe )
I think will be good some day to refactor all the tests there, in order to isolate them better )